Library/Homebrew/test/untap_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require "untap"
+
+RSpec.describe Homebrew::Untap do
+  describe ".installed_formulae_for" do
+    shared_examples "finds installed formulae in tap" do
+      def load_formula(name:, with_formula_file: false, mock_install: false)
+        formula = formula(name, tap:) do
+          url "https://brew.sh/foo-1.0.tgz"
+        end
+
+        if with_formula_file
+          class_name = name.split("_").map(&:capitalize).join
+          tap.formula_dir.mkpath
+          (tap.formula_dir/"#{name}.rb").write <<~RUBY
+            class #{class_name} < Formula
+              url "https://brew.sh/foo-1.0.tgz"
+            end
+          RUBY
+        end
+
+        if mock_install
+          keg_path = HOMEBREW_CELLAR/name/"1.2.3"
+          keg_path.mkpath
+
+          tab_path = keg_path/Tab::FILENAME
+          tab_path.write <<~JSON
+            {
+              "source": {
+                "tap": "#{tap}"
+              }
+            }
+          JSON
```

Comment:
Installing a formula is not the same as an integration test, it's just that we mainly use `install_test_formula` in integration tests.

We already have a bunch of unit tests that install casks.

Reply:
I meant more that it requires the `:integration_test` setup to use it so I assumed that it was slow. It turns out that `setup_test_formula` is quite fast and `install_test_formula` is quite slow.

Reply:
And another thing. We actually mock this install logic already in a handful of places throughout the test suite. I just think I'm the first one to actually write the tab instead of creating an empty file.

Maybe it'd be helpful to have a `mock_keg` helper or something similar.